### ISSUE TITLE
State of CSS 2025 Suggestions

### ISSUE THREAD

**Issue by SachaG** [Devographics/surveys#249]

**Comment by jsnkuhn** [comment]
I've noticed over the last few years that there are a fair number of folks who claim to be using the CSS `image()` function (https://www.w3.org/TR/css-images-4/#image-notation) even though this is not implemented in any browsers. What's the requirement for properties to get listed as features on the survey? Do they have to be implemented in at least one browser?

screenshot from 2023:

![image](https://github.com/user-attachments/assets/34b04a1b-b89d-4267-bc44-8e511f9c90d6)

**Comment by SachaG** [comment]
@jsnkuhn good point! The reason we include features that haven't been implemented yet is so that we're able to track things like awareness or interest over their entire lifetime, from conception to implementation. Also, sometimes people might've used something via a polyfill or preprocessor. In this specific case though it might very well be people just mistaking this for another feature…

**Comment by SachaG** [comment]
- Alt text for generated content

**Comment by SachaG** [comment]
- field-sizing

**Comment by SachaG** [comment]
- @import

**Comment by SachaG** [comment]
-calc-size(auto) https://blog.webdevsimplified.com/2024-07/css-calc-size/

**Comment by jsnkuhn** [comment]
From some folks over in Kevin Powell's discord:

There were some newer folks who didn't recognize many of the properties on the survey and felt the "Well Done!" was not appropriate.

There were also some folks who did better but still got the "you are in the top 100%" message? Wondering if there was a bug that showed peeps as in the top 100% too often?

![image](https://github.com/user-attachments/assets/7b46fe5f-dd66-497b-bb8a-4bffea240d9b)

![image](https://github.com/user-attachments/assets/b405a679-3183-4826-ba2c-db8ca101d1e5)

**Comment by SachaG** [comment]
Yes, the percentage calculation is (was?) buggy. I might remove it altogether, it's meant to be a fun datapoint, not something people actually care about – how many features you know or don't is pretty meaningless in the grand scheme of things…

**Comment by SachaG** [comment]
- sibling-index/sibling-count https://www.w3.org/TR/css-values-5/#tree-counting

**Comment by SachaG** [comment]
- Add pain points to each section
- Add new "CSS challenges" multiple-choice quiz section (with help from https://css-challenges.com/ ?)

**Comment by SachaG** [comment]
Some other resources for new features inspiration:

- [New CSS features to know for 2025](https://www.youtube.com/watch?v=jSCgZqoebsM
)
- https://web-platform-dx.github.io/web-features-explorer/newly-available/

**Comment by SachaG** [comment]
Ask about awareness of Baseline

**Comment by SachaG** [comment]
- `::target-text`
- ruby-align and ruby-position 
- `transition-behavior`
- `@function` - https://css-tricks.com/functions-in-css/
- `text-box-trim`
- `display: contents`

**Comment by SachaG** [comment]
- question about masonry?

**Comment by SachaG** [comment]
Preview survey here: https://github.com/Devographics/surveys/issues/289
